aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1602,6 +1602,28 @@ Tensor alias_with_sizes_and_strides(
   return self_;
 }
 
+// overload for symbolic shapes and strides.
+template <>
+Tensor alias_with_sizes_and_strides(
+    const Tensor& self,
+    const c10::SymIntArrayRef& sizes,
+    const c10::SymIntArrayRef& strides) {
+  //caller should make sure that sizes and strides are valid for self
+  //(storage is sufficient, strides are non-negative, strides and sizes array size is the same)
+  Tensor self_;
+  if (self.is_quantized()) {
+    self_ = at::detail::make_tensor<QTensorImpl>(
+      c10::TensorImpl::VIEW, Storage(self.storage()), self.key_set(), self.dtype(), get_qtensorimpl(self)->quantizer());
+    self_.unsafeGetTensorImpl()->set_sizes_and_strides(sizes, strides, self.sym_storage_offset());
```

Comment:
Is there a reason why we can't change the `ArrayRef<int64_t> and SmallVector<int64_t>` template above to use this line?

Reply:
I just forgot about that! Thanks!

Reply:
Actually, that won't work because there are two overloads for 

1. void set_sizes_and_strides(SymIntArrayRef sizes, SymIntArrayRef strides, optional<c10::SymInt> storage_offset)
2. void set_sizes_and_strides(IntArrayRef new_size, IntArrayRef new_stride, optional<int64_t> storage_offset)

So, all inputs have to be symbolic or concrete, we can't pass `self.sym_storage_offset()` to second overload.